packages/database/src/models/MedicationAdministrationRecord.ts

```
@@ -242,6 +262,129 @@ export class MedicationAdministrationRecord extends Model {
     }
   }
 
+  static async createMedicationDueTaskForMar(mar: MedicationAdministrationRecord) {
+    const { models } = this.sequelize;
+    const { Task, Encounter, EncounterPrescription, Prescription } = models;
+
+    const prescription = await Prescription.findByPk(mar.prescriptionId);
+
+    // Skip if this is a PRN medication
+    if (!prescription || prescription.isPrn) return;
+
+    const encounterPrescription = await EncounterPrescription.findOne({
+      where: { prescriptionId: prescription.id },
+      include: [
+        {
+          model: Encounter,
+          as: 'encounter',
+          attributes: ['id', 'encounterType', 'endDate'],
+        },
+      ],
+    });
+    const encounter = encounterPrescription?.encounter;
+
+    // Skip if this is not an inpatient encounter or is discharged
+    if (!encounter || encounter.encounterType !== ENCOUNTER_TYPES.ADMISSION || encounter.endDate)
```

This and the prescription equivalent [above](https://github.com/beyondessential/tamanu/pull/8013/files#diff-1c26a0c89b35775f89438f7f9624cb3178a31d9331cb8e3bb8ff0c08e921e52eR272) seem unnecessarily defensive to me. If we're inside this function, we expect both a prescription and an encounter to exist, right? So I suggest using assertions that throw errors rather than conditional checks that silently swallow.

---

it's absolutely right